### PR TITLE
[Dashboard] Data API on/off switch

### PR DESCRIPTION
**Pull request by MildTomato**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

- Data API can now be switched off easily.

Screenshot 2024-05-30 at 6 54 35 PM
——

Screenshot 2024-05-30 at 6 54 38 PM

### CODE REVIEW

**Comment by joshenlim** [comment]
@MildTomato i'm seeing the same height issue as per the previous PR that this PR was going into
image

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx [hunk 1]
```diff
@@ -34,45 +46,60 @@ import {
...
+      .optional()
+      .nullable(),
+    enableDataApi: z.boolean(),
+  })
+  .refine(
+    (data) => {
+      if (data.enableDataApi && data.dbSchema.length === 0) {
+        return false
+      }
+      return true
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:67]: nit: can be simplified

```suggestion
      return !(data.enableDataApi && data.dbSchema.length === 0)
```

**Comment by MildTomato** [comment]
> @MildTomato i'm seeing the same height issue as per the previous PR that this PR was going into

no shouldn't be

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx [hunk 2]
```diff
@@ -115,34 +127,307 @@ const PostgrestConfig = () => {
...
-        maxRows: config.max_rows,
-        dbExtraSearchPath: config.db_extra_search_path,
-        dbPool: config.db_pool,
+        ...defaultValues,
       })
     }
   }
 
+  const onSubmit = async (values: z.infer<typeof formSchema>) => {
+    if (!projectRef) return console.error('Project ref is required') // is this needed ?
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:154]: the console.error is technically not needed, just minimally the return
its just the type checking from the mutation that expects projectRef to not be undefined

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx [hunk 3]
```diff
@@ -115,34 +127,307 @@ const PostgrestConfig = () => {
...
-        <FormPanel
-          disabled={true}
-          header={<p>Data API Settings</p>}
-          footer={
-            <div className="flex px-8 py-4">
+        <FormPanelContainer>
+          <FormPanelHeader>Data API Settings</FormPanelHeader>
+          <FormPanelContent>
+            {isError ? (
+              <Admonition type="destructive" title="Failed to retrieve API settings" />
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:190]: just want to double check - should we use Admonition or AlertError
or update AlertError to use Admonition?

The latter has the cta to contact support and also surfaces the error from the network request to the user - i reckon we use AlertError?

Comment (MildTomato) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:190]: Admonition is a wrapper of Alert. same thing

Comment (joshenlim) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:190]: i'm referring to the dashboard component <AlertError> though, not the UI component <Alert_Shadcn>

Comment (MildTomato) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:190]: ah, actually your right. I'll revert it

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx (hunk 1 above)
Comment (joshenlim) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:67]: not particular on this one - no worries!

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx (hunk 3 above)
Comment (MildTomato) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:190]: sorry bit lost, what do you mean AlertError has a cta ?

**Comment by MildTomato** [comment]
currently double checking the issue with the height

**Comment by MildTomato** [comment]
fixed issue with height.

hidden <input/> fields inserted by react-hook-form were not being handled well.
we have a lot of possibly [clashing CSS](https://github.com/supabase/supabase/tree/chore/api-settings-enabled-switch/apps/studio/styles) here that could do with ripping out in future to avood this stuff.

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx (hunk 2 above)
Comment (MildTomato) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:154]: this was here before hand, it just got moved.
can we do in another PR so we don't add another point of failure ?

File: apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx (hunk 3 above)
Comment (MildTomato) [reply apps/studio/components/interfaces/Settings/API/PostgrestConfig.tsx:194]: updated to Alert

**Review by joshenlim — APPROVED** [review]

**Pull request merged**